eventlogs/postgres/events.go

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) Abstract Machines
+// SPDX-License-Identifier: Apache-2.0
+
+package postgres
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/absmach/magistrala/eventlogs"
+	"github.com/absmach/magistrala/internal/postgres"
+	"github.com/absmach/magistrala/pkg/errors"
+	repoerr "github.com/absmach/magistrala/pkg/errors/repository"
+)
+
+type repository struct {
+	db postgres.Database
+}
+
+func NewRepository(db postgres.Database) eventlogs.Repository {
+	return &repository{db: db}
+}
+
+func (repo *repository) Save(ctx context.Context, event eventlogs.Event) (err error) {
+	q := `INSERT INTO events (id, operation, occurred_at, payload)
+		VALUES (:id, :operation, :occurred_at, :payload)`
+
+	dbEvent, err := toDBEvent(event)
+	if err != nil {
+		return errors.Wrap(repoerr.ErrCreateEntity, err)
+	}
+
+	if _, err = repo.db.NamedExecContext(ctx, q, dbEvent); err != nil {
+		return postgres.HandleError(repoerr.ErrCreateEntity, err)
+	}
+
+	return nil
+}
+
+func (repo *repository) RetrieveAll(ctx context.Context, page eventlogs.Page) (eventlogs.EventsPage, error) {
+	query := pageQuery(page)
+
+	sq := "id, operation, occurred_at"
+	if page.WithPayload {
+		sq += ", payload"
+	}
+	q := fmt.Sprintf("SELECT %s FROM events %s ORDER BY occurred_at LIMIT :limit OFFSET :offset;", sq, query)
```

Comment:
we should display events from the latest. Order by `DESC`

Reply:
Added direction as a query parameter with the default being descending.